src/main/scala/org/deepdive/ddlog/DeepDiveLogCompiler.scala

```diff
@@ -182,6 +188,46 @@ class CompilationState( statements : DeepDiveLog.Program, config : DeepDiveLog.C
     }
   }
 
+  // resolve an expression
+  def resolveExpr(e: Expr, cq: ConjunctiveQuery, alias: AliasStyle, index : Int, isHead: Boolean) : String = {
```

Comment:
I think the name prefix `resolve` doesn't mean anything here (actually all the resolve methods).  Let's just be explicit about what it's doing, what kind of String it's returning.  How about `compileSQLExpr(...)` instead of `resolveExpr(...)`?